### PR TITLE
Migrate Movement 1 (Broadcast-1, Concat-1, Tile-1)

### PR DESCRIPTION
**Pull request by steve-y**
### Details:
 - *Create template plugin reference tests for target operation: Broadcast-1, Concat-1, Tile-1*
 - *Remove ngraph backend files*
 - *Update CMakeLists.txt*
 - *Add R"(.\*ReferenceConcatTest.\*concat_zero_.\*)" to skip_tests_config.cpp*

### Tickets:
 - *64008*
 - *63947*
 - *64000*

### CODE REVIEW

**Comment by steve-y** [comment]
@iefode @ilyachur skip_tests_config.cpp contains R"(.*ReferenceConcatTest.*concat_zero_.*)" because Concat generated the following exception:
```
C++ exception with description "
cnn_network_ngraph_impl.cpp:75 Parameter_123166 has zero dimension which is not allowed" thrown in the test body.
```
In the following tests:
```
concat_zero_length_1d_last
concat_zero_length_1d_middle
concat_zero_zero
concat_zero_length_4d_middle
```
They seem to have zero dimension test.

File: docs/template_plugin/tests/functional/op_reference/concat.cpp [hunk 1]
```diff
@@ -0,0 +1,1436 @@
...
+            result << "_rShape=" << param.expected.shape;
+        }
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const ConcatParams& params) {
+        std::shared_ptr<op::v0::Parameter> A, B, C;
+        if (params.dynamicShape.is_dynamic()) {
+            A = std::make_shared<op::v0::Parameter>(params.A.type, params.dynamicShape);
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/concat.cpp:67]: Could we use `ov::opsetN::` istead of `op::vN`?

Comment (steve-y) [reply docs/template_plugin/tests/functional/op_reference/concat.cpp]: Updated.

**Review by iefode — APPROVED** [review]

File: docs/template_plugin/tests/functional/op_reference/broadcast.cpp [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+        actualOutData.clear();
+        for (const auto &output : executableNetwork.outputs()) {
+            actualOutData.emplace_back(inferRequest.get_tensor(output));
+        }
+        refOutData = actualOutData;
+    }
+};
+
+TEST_P(ReferenceBroadcastTestTestHelper, CompareWithRefs) {
+    LoadNetwork();
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/broadcast.cpp:219]: Why do we need a custom pipeline here?

Comment (steve-y) [reply docs/template_plugin/tests/functional/op_reference/broadcast.cpp:219]: > Why do we need a custom pipeline here?

The broadcast backend test has `broadcast_test_helper`. It calls the test function two times. And the second function uses the output of the first function.
So this test needs output without validation. Do we have any methods in this kind of test?

**Review by ilyachur — APPROVED** [review]

**Pull request merged**